homeassistant/components/sensor/rfxtrx.py

```diff
@@ -46,13 +47,16 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
                 if data_type in event.values:
                     data_types = [data_type]
                     break
+        add_devices_callback(sensors)
         for _data_type in data_types:
-            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME],
-                                      _data_type, entity_info[ATTR_FIREEVENT])
-            sensors.append(new_sensor)
+            entity_id = generate_entity_id(ENTITY_ID_FORMAT,
+                                           entity_info[ATTR_NAME],
+                                           hass=hass)
+            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME], _data_type,
+                                      entity_info[ATTR_FIREEVENT], entity_id)
             sub_sensors[_data_type] = new_sensor
+            add_devices_callback([new_sensor])
```

Comment:
Okay, my bad again 😞 . You are right.

We are adding all entities as tasks [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_component.py#L316-L319).

We should move those to be done in sync instead.

CC @pvizeli